src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleDictionary.cs

```diff
@@ -156,5 +152,22 @@ public static partial class JsonSerializer
 #endif
             }
         }
+
+        private static void WriteExtensionData(Utf8JsonWriter writer, ref WriteStackFrame frame)
+        {
+            JsonPropertyInfo jsonPropertyInfo = frame.JsonPropertyInfo;
+
+            DictionaryEntry entry = ((IDictionaryEnumerator)frame.Enumerator).Entry;
+            if (entry.Value is JsonElement)
+            {
+                Debug.Assert(entry.Key is string);
```

Comment:
What's guaranteeing this? Shouldn't it be something like:
```C#
if (entry.Value is JsonElement element && entry.Key is string propertyName)
{
   element.WriteAsProperty(propertyName, writer);
}
```

Reply:
We previously verified the extension data property TKey is type `string`. The TValue however may be `Object` or `JsonElement`

Reply:
> We previously verified the extension data property TKey is type `string`.

Where? Can we defer that check to this point instead or do we need to verify somewhere earlier up the call stack?